### PR TITLE
Ignore multi-line git trailers in configure_ci.py.

### LINKED ISSUE #19240: configure_ci.py interprets `Notes:` as "trailer lines"
All CI jobs on https://github.com/iree-org/iree/pull/19194 were failing since I wrote a nice long description. I made another PR to debug that, https://github.com/iree-org/iree/pull/19239, and what I found is that having a PR description where one line is just

```
Notes:
```

gets interpreted by `configure_ci.py` as `trailer_lines`, which it then tries to split as `key:value` and that complains that there is only one term here, not two.

See the CI log here; that PR adds a `print(trailer_lines)` so we see that it starts with `"Notes:"`:
https://github.com/iree-org/iree/actions/runs/11945467240/job/33298195480#step:4:94

Notice the difference between the PR descriptions of  https://github.com/iree-org/iree/pull/19239 (failing) and https://github.com/iree-org/iree/pull/19194 (working): in the latter, I removed the `:` after `Notes` .

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/iree-org/iree/issues/19240 (ugh string parsing).

### CODE REVIEW

File: build_tools/github_actions/configure_ci.py [hunk 1]
```diff
@@ -224,6 +224,32 @@ def check_description_and_show_diff(
...
+        timeout=60,
+    ).stdout.splitlines()
+
+    # Skip over multi-line or malformed trailers we don't want to support.
+    # https://github.com/iree-org/iree/issues/19240
+    # https://stackoverflow.com/q/66215644
+    # We could also handle multi-line git trailers, but we'd need to rework the
+    # .splitlines() call above.
+    trailer_lines = [line for line in trailer_lines if not line.startswith((" ", "\t"))]
+    trailer_lines = [line for line in trailer_lines if ":" in line]
```
Comment (bjacob) [inline build_tools/github_actions/configure_ci.py:243]: Does this need to also require that the `:` occurs in the middle of the string, not at the start or at the end as in `Notes:` ? I thought that played a role in `line.split(":")` returning a single element and failing the requirement of the structured binding to a pair `k, v` ?

Comment (ScottTodd) [reply build_tools/github_actions/configure_ci.py:243]: The lines there are doing the same thing. I added both for some safety / redundancy in case I'm missing some edge case.

Some printf debugging of the test cases:

### test_parse_trailer_map_multiline_trailer

```
First line

key: value line 1
  value line 2
```

Parsed by `git interpret-trailers` --> `.splitlines()` as
```
['key: value line 1', ' value line 2']
```

Either list comprehension will remove the second element of that list, which fails at the split -> unpack code.

### test_parse_trailer_map_multiline_trailer_skip_first

```
First line

key:
  value line 2
  value line 3
```

Parsed by `git interpret-trailers` --> `.splitlines()` as

```
['key: value line 2', ' value line 3']
```

**Review by bjacob — APPROVED** [review]

File: build_tools/github_actions/configure_ci.py (hunk 1 above)
Comment (ScottTodd) [reply build_tools/github_actions/configure_ci.py:243]: The combination of multi-line git trailers being a thing and our code using `splitlines()` to parse the stdout output of `git interpret-trailers` is what is causing problems.

Consider this case:

```
First line

key1:
  value1 line 2
  value1 line 3
key2: value2
```

This is what we get:
```
['key1: value1 line 2', ' value1 line 3', 'key2: value2']
```

When this is what we might want instead:
```
['key1: value1 line 2\n value1 line 3', 'key2: value2']
```

Some string escaping could help, or finding a better way to parse the multi-line trailers. IDK, I don't think we really need to support that case.
